src/com/owncloud/android/services/observer/FileObserverService.java

```diff
@@ -338,7 +382,46 @@ private void removeObservedFile(String localPath) {
         }
     }
 
-    
+    /**
+     * Requests the update of the observers responsible to watch folders where new files
+     * should be automatically added to OC, according to changes in instant upload settings.
+     */
+    private void updateInstantUploadsObservers() {
+        PreferenceManager.InstantUploadsConfiguration config =
+            PreferenceManager.getInstantUploadsConfiguration(this);
+
+        if ((   config.isEnabledForPictures() || config.isEnabledForVideos()) &&
+                mInstantUploadsObserver == null
+            )  {
+            // no current observer -> create it
+            mInstantUploadsObserver = new InstantUploadsObserver(config, getApplicationContext());
+            mInstantUploadsObserver.startWatching();
+
+        } else if ( !config.isEnabledForPictures() && !config.isEnabledForVideos() &&
+                    mInstantUploadsObserver != null
+            ) {
+            // nothing ot observe -> stop current observer
+            mInstantUploadsObserver.stopWatching();
+            mInstantUploadsObserver = null;
+
+        } else if ( mInstantUploadsObserver != null &&
+                    !mInstantUploadsObserver.getSourcePath().equals(config.getSourcePath())
+            ) {
+            // source path to watch was changed -> stop current observer, create a new one
+            mInstantUploadsObserver.stopWatching();
+            mInstantUploadsObserver = new InstantUploadsObserver(config, getApplicationContext());
+            mInstantUploadsObserver.startWatching();
+
+        } else if ( mInstantUploadsObserver != null) {
+            // observer exists and can handle changes in configuration -> update observer
+            mInstantUploadsObserver.updateConfiguration(config);
+
+        } else {
+            Log_OC.w(TAG, "No current observer for instant uploads, nothing to observe in configuration");
+            // this should not happen, but it's not dangerous
```

Comment:
Comment is wrong, this will happen when instant uploads are disabled at any time that the service is initialized (for instance: in BOOT time or when the OC app is started).

Log should be ```info```, not level.